app/controllers/news_letter_controller.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class NewsLetterController < ApplicationController
+  def index
+    validate_news_letter_email_is_valid
+    GoogleSpreadSheet.new.create_news_letter_email_in_google_sheets(params)
+    respond_to do |format|
+      format.html 
+      format.js 
+    end
+    flash[:notice] = "Thanks for subscribing! You will receive an e-mail confirmation shortly."
+  rescue StandardError => e
+    flash[:error] = e.message
+  end
+
+
+  def validate_news_letter_email_is_valid
```

Comment:
Maybe a quickie one-liner doc string on this.

Reply:
Do you really want to allow test@com?  When you say "news-letter email", do you mean the email address a teacher will sign up for the newsletter with?  Or something else?

Reply:
@daryanypl Yes. I didn't write any custom regular expression for email validation. Stephen asked me use ruby email validation. Check here this is ruby method.  gem 'email_validator',  EmailValidator.valid? params['email']

Reply:
ok